Log enhancement when flash.service_addr is wrong

## Enhancement
If the user deploy TiFlash with old version TiDB Operator and not follow the [step](https://docs.pingcap.com/zh/tidb-in-kubernetes/v1.1/deploy-tiflash#%E4%B8%8D%E5%90%8C%E7%89%88%E6%9C%AC%E9%85%8D%E7%BD%AE%E6%B3%A8%E6%84%8F%E4%BA%8B%E9%A1%B9)  to manually modify the `flash.service_addr` configuration, GRPC server will bind fail and cause TiFlash segment fault, might be confusing for users as `grpc server` is not mentioned in documentation, and user may don't know which address is wrong.


[2022/06/29 11:34:07.231 +08:00] [ERROR] [Server.cpp:308] ["grpc:/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tics/contrib/grpc/src/core/ext/transport/chttp2/server/insecure/server_chttp2.cc, line number: 40, log msg : {\"created\":\"@1656473647.231704483\",\"description\":\"No address added out of total 1 resolved\",\"file\":\"/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tics/contrib/grpc/src/core/ext/transport/chttp2/server/chttp2_server.cc\",\"file_line\":395,\"referenced_errors\":[{\"created\":\"@1656473647.231702053\",\"description\":\"Unable to configure socket\",\"fd\":36,\"file\":\"/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tics/contrib/grpc/src/core/lib/iomgr/tcp_server_utils_posix_common.cc\",\"file_line\":208,\"referenced_errors\":[{\"created\":\"@1656473647.231697730\",\"description\":\"Cannot assign requested address\",\"errno\":99,\"file\":\"/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tics/contrib/grpc/src/core/lib/iomgr/tcp_server_utils_posix_common.cc\",\"file_line\":181,\"os_error\":\"Cannot assign requested address\",\"syscall\":\"bind\"}]}]}"] [thread_id=1]
[2022/06/29 11:34:07.232 +08:00] [ERROR] [BaseDaemon.cpp:377] [BaseDaemon:########################################] [thread_id=5]
[2022/06/29 11:34:07.232 +08:00] [ERROR] [BaseDaemon.cpp:378] ["BaseDaemon:(from thread 1) Received signal Segmentation fault(11)."] [thread_id=5]
[2022/06/29 11:34:07.232 +08:00] [ERROR] [BaseDaemon.cpp:408] ["BaseDaemon:Address: 0x7f688c270000"] [thread_id=5]
[2022/06/29 11:34:07.232 +08:00] [ERROR] [BaseDaemon.cpp:414] ["BaseDaemon:Access: read."] [thread_id=5]
[2022/06/29 11:34:07.232 +08:00] [ERROR] [BaseDaemon.cpp:420] ["BaseDaemon:Attempted access has violated the permissions assigned to the memory area."] [thread_id=5]
[2022/06/29 11:34:10.125 +08:00] [ERROR] [BaseDaemon.cpp:570] ["BaseDaemon:\n       0x65913b1\tfaultSignalHandler(int, siginfo_t*, void*) [tiflash+106501041]\n                \tlibs/libdaemon/src/BaseDaemon.cpp:221\n  0x7f68a2005630\t<unknown symbol> [libpthread.so.0+63024]\n       0x7ed1dc0\tgrpc_server_request_registered_call [tiflash+132980160]\n                \tcontrib/grpc/src/core/lib/surface/server.cc:0\n       0x7584c38\tDB::EstablishCallData::EstablishCallData(DB::AsyncFlashService*, grpc_impl::ServerCompletionQueue*, grpc_impl::ServerCompletionQueue*, std::__1::shared_ptr<std::__1::atomic<bool> > const&) [tiflash+123227192]\n                \tdbms/src/Flash/EstablishCall.cpp:34\n       0x1d434ac\tDB::Server::main(std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) [tiflash+30684332]\n                \tdbms/src/Server/Server.cpp:1422\n       0x7959d9a\tPoco::Util::Application::run() [tiflash+127245722]\n                \tcontrib/poco/Util/src/Application.cpp:335\n       0x7960d84\tPoco::Util::ServerApplication::run(int, char**) [tiflash+127274372]\n                \tcontrib/poco/Util/src/ServerApplication.cpp:618\n       0x1d4e74b\tmainEntryClickHouseServer(int, char**) [tiflash+30730059]\n                \tdbms/src/Server/Server.cpp:1570\n       0x1ccc7fc\tmain [tiflash+30197756]\n                \tdbms/src/Server/main.cpp:179\n  0x7f68a1a33555\t__libc_start_main [libc.so.6+140629]"] [thread_id=5]

--- a/dbms/src/Server/Server.cpp
+++ b/dbms/src/Server/Server.cpp
@@ -625,7 +625,7 @@ public:
         flash_grpc_server = builder.BuildAndStart();
         if (!flash_grpc_server)
         {
-            throw Exception("Exception happens when start grpc server, the flash.service_addr may be wrong, flash.service_addr is " + raft_config.flash_server_addr, ErrorCodes::IP_ADDRESS_NOT_ALLOWED);
+            throw Exception("Exception happens when start grpc server, the flash.service_addr may be invalid, flash.service_addr is " + raft_config.flash_server_addr, ErrorCodes::IP_ADDRESS_NOT_ALLOWED);
         }
         LOG_FMT_INFO(log, "Flash grpc server listening on [{}]", raft_config.flash_server_addr);
         Debug::setServiceAddr(raft_config.flash_server_addr);

--- a/dbms/src/Server/Server.cpp
+++ b/dbms/src/Server/Server.cpp
@@ -185,6 +185,7 @@ extern const int NO_ELEMENTS_IN_CONFIG;
 extern const int SUPPORT_IS_DISABLED;
 extern const int ARGUMENT_OUT_OF_BOUND;
 extern const int INVALID_CONFIG_PARAMETER;
+extern const int IP_ADDRESS_NOT_ALLOWED;
 } // namespace ErrorCodes
 
 namespace Debug
@@ -622,6 +623,10 @@ public:
             }
         }
         flash_grpc_server = builder.BuildAndStart();
+        if (!flash_grpc_server)
+        {
+            throw Exception("Exception happens when start grpc server, the flash.service_addr may be wrong, flash.service_addr is " + raft_config.flash_server_addr, ErrorCodes::IP_ADDRESS_NOT_ALLOWED);
+        }
         LOG_FMT_INFO(log, "Flash grpc server listening on [{}]", raft_config.flash_server_addr);
         Debug::setServiceAddr(raft_config.flash_server_addr);
         if (enable_async_server)

